zipline/utils/serialization_utils.py

```diff
@@ -0,0 +1,55 @@
+# Label for the serialization version field in the state returned by
+# __getstate__.
+VERSION_LABEL = '_stateversion_'
+
+
+class SerializeableZiplineObject(object):
+    """
+    This class implements the basic set and get state methods used for
+    serialization. It also serves as a demarkation of which objects we
+    serialize.
+    """
+
+    def __getstate__(self):
+        """
+        Many get_state methods need this one line of code.
+        This method deduplicates the code calls.
+        """
+        state_dict = \
+            {k: v for k, v in self.__dict__.iteritems()
+                if not k.startswith('_')}
+        return state_dict
+
+    def __setstate__(self, state):
+        """
+        Many objects require only this code.
+        """
+        self.__dict__.update(state)
+
+    # =====================================================
+    # These are helper methods for some problem data types.
+    # =====================================================
+
```

Comment:
Something like this: https://github.com/quantopian/zipline/pull/505/files#diff-f3d976ceea2b1ea116269ac8b7e9ccc8R428

These objects needs to be internally consistent. These objects won't inflate without the internal stuff that knows what those special types mean.